File: 1400/489C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int m, s;
    cin >> m >> s;

    int s1 = s - 1;

    // Impossible cases
    if ((s == 0 && m != 1) || (m * 9 < s)) 
    {
        cout << "-1 -1\n";
        return 0;
    }

    string ans1 = "", ans2 = "";

    // Construct maximum number (ans1)
    int temp = s;
    for (int i = 0; i < m; i++) 
    {
        int digit = min(9, temp);
        ans1 += char('0' + digit);
        temp -= digit;
    }

    // Construct minimum number (ans2)
    temp = s1;
    for (int i = 0; i < m - 1; i++) 
    {
        int digit = min(9, temp);
        ans2 = char('0' + digit) + ans2;
        temp -= digit;
    }
    
    ans2 = char('0' + (temp + 1)) + ans2;

    cout << ans2 << " " << ans1 << "\n";
    return 0;
}

File: 1400/489C/main.java
import java.util.*;
public class main
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        
        int m = sc.nextInt();
        int s = sc.nextInt();
        
        int[] arr = new int[m];
        int s1 = s-1;
        
        if(s==0 && m!=1 || m*9<s)
        {
            System.out.println("-1 -1");
            return;
        }
        
        String ans1 = "", ans2 = "";
        for(int i=0;i<m;i++)
        {
            ans1 += Math.min(9, s);
            s -= Math.min(9, s);
        }
        
        for(int i=0;i<m-1;i++)
        {
            ans2 = Math.min(9, s1) + ans2;
            s1 -= Math.min(9, s1);
        }
        
        ans2 = (s1+1) + ans2;

        System.out.println(ans2 + " " + ans1);
    }
}